services/route_manager/src/main/java/com/futurewei/alcor/route/service/Impl/RouterServiceImpl.java

```diff
@@ -281,17 +282,49 @@ public RouteTable getSubnetRouteTable(String projectId, String subnetId) throws
     }
 
     @Override
-    public RouteTable updateSubnetRouteTable(String projectId, String subnetId, RouteTableWebJson resource) throws CacheException, DatabasePersistenceException, OwnMultipleSubnetRouteTablesException, CanNotFindVpc, CanNotFindSubnet, ResourceNotFoundException, ResourcePersistenceException, OwnMultipleVpcRouterException {
-        RouteTable routeTable = new RouteTable();
-        RouteTable inRoutetable = resource.getRoutetable();
+    public RouteTable updateSubnetRouteTable(String projectId, String subnetId, UpdateRoutingRuleResponse resource) throws CacheException, DatabasePersistenceException, OwnMultipleSubnetRouteTablesException, CanNotFindVpc, CanNotFindSubnet, ResourceNotFoundException, ResourcePersistenceException, OwnMultipleVpcRouterException {
+        InternalSubnetRoutingTable inRoutetable = resource.getInternalSubnetRoutingTable();
+        List<HostRoute> inHostRoutes = resource.getHostRouteToSubnet();
         // Get or create a router for a Subnet
-        routeTable = getSubnetRouteTable(projectId, subnetId);
+        RouteTable routeTable = getSubnetRouteTable(projectId, subnetId);
+
+        for (InternalRoutingRule inRule : inRoutetable.getRoutingRules()) {
+            if (OperationType.CREATE.equals(inRule.getOperationType())) {
+                RouteEntry newRoute = new RouteEntry();
+                newRoute.setId(inRule.getId());
+                newRoute.setName(inRule.getName());
+                newRoute.setDestination(inRule.getDestination());
+                newRoute.setNexthop(inRule.getNextHopIp());
+                newRoute.setPriority(inRule.getPriority());
+                routeTable.getRouteEntities().add(newRoute);
```

Comment:
@cj-chung We see multiple times of converting one entity from one form to the other. Let us write some method or even define converting class for this purpose.

Reply:
The set codes are replaced by the RouteEntry constructor.